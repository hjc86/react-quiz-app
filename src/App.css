* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: 'Exo', sans-serif;
  font-weight: 100px;
}

.container {
  display: flex;
  flex-direction: column;
  
}