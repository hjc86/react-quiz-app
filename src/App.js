import React from 'react';
import UserInfo from './components/UserInfo';
import Quiz from './components/Quiz';
import Score from './components/Score';
import './App.css';


class App extends React.Component {

// states GetUsername, GetAPI, GetQuestions, GetAnswers, GetScore
constructor(props){
  super(props);
  this.state = {
      QuizQ : [],
      Answers : [],
      userName: null,
      difficulty: null,
      category: null
  };
}

// Fetch API
getQuizApi = async (userData) => {

  const categoryNumber = userData.category;
  const difficulty = userData.difficulty;
  const apiCall = await fetch(`https://opentdb.com/api.php?amount=5&category=${categoryNumber}&difficulty=${difficulty}&type=multiple`);
  const data = await apiCall.json();

  console.log(userData);
 
  // to get the whole data 
  this.setState({
    QuizQ : data.results
  
  });
  console.log(data);
}


getUserData =(userDataObj) =>{
  this.setState(
    userDataObj
  ) 
  // console.log("before update",this.state)
}

componentDidUpdate(){
  // console.log("after update",this.state)
}


formSubmitted = (state) => {
  this.setState({
      userName: state.username,
      difficulty: state.difficulty,
      category: state.category
  })
  console.log(state);
  this.getQuizApi(state);
  
}



    render() {
      return (
        <div className="container">
          <UserInfo userData={this.getUserData} onSubmit={this.formSubmitted}/>
          {this.state.QuizQ.map(questions => {
            const answers = [questions.correct_answer, ...questions.incorrect_answers]
            console.log(answers);
            return <Quiz questions={questions.question} answers={questions.correct_answer} incorrect={questions.incorrect_answers}/>
          })
          }
          <Score />
        </div>
      );
}

}

export default App;

// answers={questions.correct_answer}