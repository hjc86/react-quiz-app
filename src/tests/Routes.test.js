import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme'; 
import Routes from '../components/Routes';
import renderer from 'react-test-renderer';
import App from '../App'



describe('Routes', () => {
  it('mounts to DOM', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Routes />, div);
      ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
  const tree = renderer
    .create(<Routes/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

 

  let wrapper;

    /// end of setup //

    beforeEach(() => wrapper = shallow(<Routes />));

    it('should render a single <div />',()=>{
        expect(wrapper.find('div').length).toEqual(1);
    })

    it('should render 1 Route to App Component', () =>{
      expect(wrapper.find('Route').length).toEqual(1);
  })
 
})