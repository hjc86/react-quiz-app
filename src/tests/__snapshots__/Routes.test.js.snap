// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="formDiv"
    >
      <div
        className="formContainer"
      >
        <form
          onSubmit={[Function]}
        >
          <label
            className="player"
          >
            Player(s)
          </label>
           
          <br />
          <input
            className="bar"
            id="username"
            name="username"
            placeholder="insert player name(s)"
            required={true}
            type="text"
          />
          <p
            className="diff"
          >
            Select Level
          </p>
          <label
            className="text"
          >
            Easy 
          </label>
          <input
            id="easy"
            name="difficulty"
            required={true}
            type="radio"
            value="easy"
          />
          <label
            className="text"
          >
            Medium 
          </label>
          <input
            id="medium"
            name="difficulty"
            type="radio"
            value="medium"
          />
          <label
            className="text"
          >
            Difficult 
          </label>
          <input
            id="difficult"
            name="difficulty"
            type="radio"
            value="hard"
          />
          <br />
          <div
            className="categoryDiv"
          >
            <select
              id="category"
              name="category"
              required={true}
              value=""
            >
              <option
                name="Please Select Category"
                value=""
              >
                 Please Select Category
              </option>
              <option
                name="Celebrity"
                value="26"
              >
                Celebrity
              </option>
              <option
                name="Mythology"
                value="20"
              >
                Mythology
              </option>
              <option
                name="History"
                value="23"
              >
                History
              </option>
              <option
                name="Animals"
                value="27"
              >
                Animals
              </option>
              <option
                name="Science & Computers"
                value="18"
              >
                Science & Computers
              </option>
            </select>
          </div>
          <div
            className="instructions"
          >
            <span
              className="instructHead"
            >
              Player Instructions
            </span>
            <span
              className="instruct"
            >
              1 : Insert Player Name(s) add " , " for multiplayer
            </span>
            <span
              className="instruct"
            >
              2 : Each player will have a set of 5 questions 
            </span>
            <span
              className="instruct"
            >
              See who's the Whiz in this Quiz!
            </span>
            <em
              className="instruct"
            >
              Whuiz the Quiz App
            </em>
          </div>
          <div
            className="buttonDiv"
          >
            <button
              className="playbutton"
              type="submit"
            >
              Play
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
